Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/NetworkSearchController.java

```diff
@@ -131,53 +146,104 @@ public String getSearch(@ProjectIdentifier @PathVariable("projectUnixName") Stri
     }
 
     /**
-     * This method returns a display of the network that consists of all the statements in the
-     * given projects that contain the provided concept id.
-     * @param projectUnixName Unix name of the project whose networks will be searched for the concept
-     * @param conceptId Id of the concept
+     * This method accepts a request to fetch network corresponding to the
+     * searched term. The task of generating the network is delegated to a
+     * separate thread and a token (which identifies the submitted request) is
+     * generated.
+     * 
+     * @param projectUnixName
+     * @param conceptId
      * @param project
-     * @param model
      * @return relative path to the web-page that displays the network
      * @throws QuadrigaStorageException
      */
     @CheckPublicAccess
     @InjectProjectByName
-    @RequestMapping(value = "sites/{projectUnixName}/networks/search", method = RequestMethod.GET)
-    public String getSearchTransformedNetwork(@ProjectIdentifier @PathVariable("projectUnixName") String projectUnixName,
+    @RequestMapping(value = "sites/{projectUnixName}/networks/search", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public String submitSearchTransformedNetworkRequest(
+            @ProjectIdentifier @PathVariable("projectUnixName") String projectUnixName,
             @RequestParam("conceptId") String conceptId, @CheckAccess @InjectProject IProject project, Model model)
-                    throws QuadrigaStorageException {
-        
+            throws QuadrigaStorageException {
+
         String lemma = "";
         String searchNodeLabel = "";
-        
         // Fetch ConceptPower entries related to the conceptId
         IConcept concept = cpCache.getConceptByUri(conceptId);
-   
+
         if (concept != null) {
             searchNodeLabel = concept.getWord();
             lemma = concept.getDescription();
-           
         }
-     
-        ITransformedNetwork transformedNetwork = transformationManager
-                .getSearchTransformedNetwork(project.getProjectId(), conceptId, INetworkStatus.APPROVED);
 
-        String json = null;
-        if (transformedNetwork != null) {
-            json = jsonCreator.getJson(transformedNetwork.getNodes(), transformedNetwork.getLinks());
-        }
+        Callable<PublicSearchObject> callable = () -> {
 
-        if (transformedNetwork == null || transformedNetwork.getNodes().size() == 0) {
-            model.addAttribute("isNetworkEmpty", true);
+            PublicSearchObject publicSearchObject = new PublicSearchObject();
+            ITransformedNetwork transformedNetwork = transformationManager
+                    .getSearchTransformedNetwork(project.getProjectId(), conceptId, INetworkStatus.APPROVED);
```

Comment:
```getSearchTransformedNetwork``` should be called as a asycn method (```@async``` annotation). turning the resulting network into json should be done in the method that checks for the results.